### PR TITLE
Disable TensorPrimitives vectorization of Log, Cbrt, Pow, and RootN

### PR DESCRIPTION
**Pull request by stephentoub**
We had a test bug that was hiding cases where one of the expected / actual was NaN and the other wasn't.

https://github.com/dotnet/runtime/issues/100535

### CODE REVIEW

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.Log.cs [hunk 1]
```diff
@@ -76,8 +76,7 @@ public static void Log<T>(ReadOnlySpan<T> x, T y, Span<T> destination)
         internal readonly struct LogOperator<T> : IUnaryOperator<T, T>
             where T : ILogarithmicFunctions<T>
         {
-            public static bool Vectorizable => (typeof(T) == typeof(double))
-                                            || (typeof(T) == typeof(float));
+            public static bool Vectorizable => false; // typeof(T) == typeof(float) || typeof(T) == typeof(double); // TODO: https://github.com/dotnet/runtime/issues/100535
```
Comment (tannergooding) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.Log.cs:79]: What's failing for `Log`? It has a proper implementation and has tests covering all the same scenarios as scalar is testing?

The other APIs are the ones naively implemented in terms of log, and not actually checking for the relevant edge case handling.

Comment (stephentoub) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.Log.cs]: It's the overload of Log that takes a base.

Comment (tannergooding) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.Log.cs]: Thanks for confirming!

**Comment by stephentoub** [comment]
@tannergooding, mind approving if it looks ok to you?

**Review by tannergooding — APPROVED** [review]

**Comment by LoopedBard3** [comment]
Regressions seemingly due to this:
https://github.com/dotnet/perf-autofiling-issues/issues/33681

**Comment by stephentoub** [comment]
> Regressions: Linux arm64: [dotnet/perf-autofiling-issues#33681](https://github.com/dotnet/perf-autofiling-issues/issues/33681)

Thanks. This is good to see, actually; shows that the vectorization was beneficial. We just need to make it correct first :)

**Comment by EgorBo** [comment]
Same on x64:

- https://github.com/dotnet/perf-autofiling-issues/issues/33880

**Comment by DrewScoggins** [comment]
There were also some possible improvements here: https://github.com/dotnet/perf-autofiling-issues/issues/33927
